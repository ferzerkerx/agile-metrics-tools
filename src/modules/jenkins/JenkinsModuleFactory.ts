import { CollectorModuleFactory, Utils } from '../../metrics';
import { JenkinsCollectorConfig, JenkinsMetricItem } from './collector/Types';
import { JenkinsService } from './Types';
import { JenkinsClientImpl } from './JenkinsClientImpl';
import { JenkinsServiceImpl } from './JenkinsServiceImpl';
import { JenkinsCollectorsService } from './collector/JenkinsCollectorsService';
import { JenkinsRepositoryImpl } from './JenkinsRepositoryImpl';

export class JenkinsModuleFactory
  implements CollectorModuleFactory<JenkinsCollectorConfig, JenkinsMetricItem> {
  private static jenkinsService(): JenkinsService {
    Utils.checkEnvVar('JENKINS_HOST', 'JENKINS_USER', 'JENKINS_API_TOKEN');
    const jenkinsClient = new JenkinsClientImpl({
      host: `${process.env.JENKINS_HOST}`,
      apiToken: `${process.env.JENKINS_API_TOKEN}`,
      apiUser: `${process.env.JENKINS_USER}`
    });
    const jenkinsRepository = new JenkinsRepositoryImpl(jenkinsClient);
    return new JenkinsServiceImpl(jenkinsRepository);
  }

  collectorInstance(): JenkinsCollectorsService {
    return new JenkinsCollectorsService(JenkinsModuleFactory.jenkinsService());
  }

  collectorConfiguration(obj: any): JenkinsCollectorConfig {
    return new JenkinsCollectorConfig(obj);
  }
}
