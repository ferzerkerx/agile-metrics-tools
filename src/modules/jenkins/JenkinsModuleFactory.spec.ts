import { JenkinsModuleFactory } from './JenkinsModuleFactory';

describe('JenkinsModuleFactory', () => {
  const jenkinsModuleFactory: JenkinsModuleFactory = new JenkinsModuleFactory();

  beforeEach(() => {
    jest.resetModules();
    delete process.env.JENKINS_HOST;
    delete process.env.JENKINS_API_TOKEN;
    delete process.env.JENKINS_USER;
  });

  it('should create collector', () => {
    process.env.JENKINS_HOST = 'SOME_TOKEN';
    process.env.JENKINS_API_TOKEN = 'SOME_TOKEN';
    process.env.JENKINS_USER = 'SOME_TOKEN';
    const collectorInstance = jenkinsModuleFactory.collectorInstance();
    expect(collectorInstance).not.toBeNull();
  });

  it('should fail to create collector', () => {
    expect(
      jenkinsModuleFactory.collectorInstance
    ).toThrowErrorMatchingSnapshot();
  });

  it('should create collectorConfiguration', () => {
    const collectorConfiguration = jenkinsModuleFactory.collectorConfiguration(
      {}
    );
    expect(collectorConfiguration).not.toBeNull();
  });
});
