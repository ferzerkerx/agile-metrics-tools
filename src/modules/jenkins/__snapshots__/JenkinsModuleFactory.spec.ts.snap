// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JenkinsModuleFactory should fail to create collector 1`] = `"env.JENKINS_HOST not set!"`;
